spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/StoredProcExecutor.java

```diff
@@ -210,23 +233,31 @@ public void afterPropertiesSet() {
 
 		}
 
-		jdbcCallOperationsCache = CacheBuilder.newBuilder()
-				.maximumSize(jdbcCallOperationsCacheSize)
-				.recordStats()
-				.build(new CacheLoader<String, SimpleJdbcCallOperations>() {
-					@Override
-					public SimpleJdbcCall load(String storedProcedureName) {
-						return createSimpleJdbcCall(storedProcedureName);
-					}
-				});
+		if (guavaPresent) {
+			this.jdbcCallOperationsCache = CacheBuilder.newBuilder()
+					.maximumSize(this.jdbcCallOperationsCacheSize)
+					.recordStats()
+					.build(new CacheLoader<String, SimpleJdbcCallOperations>() {
+
+						@Override
+						public SimpleJdbcCall load(String storedProcedureName) {
+							return createSimpleJdbcCall(storedProcedureName);
+						}
+					});
+		}
 
 		if (this.storedProcedureNameExpression instanceof LiteralExpression) {
 			String storedProcedureName = this.storedProcedureNameExpression.getValue(String.class);
 			final SimpleJdbcCall simpleJdbcCall = createSimpleJdbcCall(storedProcedureName);
-			this.jdbcCallOperationsCache.put(storedProcedureName, simpleJdbcCall);
+			if (guavaPresent) {
+				this.jdbcCallOperationsCache.put(storedProcedureName, simpleJdbcCall);
+			}
+			else {
+				this.jdbcCallOperationsMap.put(storedProcedureName, simpleJdbcCall);
```

Comment:
Should synchronize here too, to avoid Sonar barking at us (inconsistent sync).

That said, it's not clear to me why we preload this one.